### PR TITLE
Explicitly use the alloc op context for dpd nat updates

### PR DESCRIPTION
**Pull request by zephraph**
There was an issue in main where deleting an instance failed w/ a 403. Digging in further it looked like the 403 was coming from the instance delete saga, specifically the `InstanceDeleteNat` step that was added in #4610. It turns out that called `instance_delete_dpd_config` which called `boundary_switches` with the normal opcontext. Boundary switches calls `list_switch_ports_with_uplinks` and that requires fleet level permissions. 

The fix here was to pass the `opcontext_alloc` which gives higher level permissions down the `boundary_switches` call chain. 

I believe that we should _ultimately_ try to figure out some way to encode our authz state into the type system such that you can't just pass down a generic opcontext. I'm not exactly sure _how_ we would do that, but it seems worth pursuing.

### CODE REVIEW

**Comment by zephraph** [comment]
I think there's a test failure here. If anyone feels like getting this over the line, please, be my guest.

**Comment by zephraph** [comment]
@jmpesp I appreciate you!

File: nexus/src/app/instance_network.rs [hunk 1]
```diff
@@ -438,6 +438,7 @@ impl super::Nexus {
     pub(crate) async fn instance_delete_dpd_config(
         &self,
         opctx: &OpContext,
+        opctx_alloc: &OpContext,
```
Comment (smklein) [inline nexus/src/app/instance_network.rs:441]: I think the usage of this `opctx_alloc` is correct - as in, it appears to be the right opctx we should use -- but this is a method of the `Nexus ` struct which contains the `opctx_alloc` field already (see: `nexus/src/app/mod.rs` for the definition).

Could we just access `self.opctx_alloc`? Seems weird to have it be passed as a parameter, if we don't really want the caller to customize it.

Comment (zephraph) [reply nexus/src/app/instance_network.rs:441]: Oh! Yes, that would make sense.

**Review by smklein — APPROVED** [review]
LGTM, only caveat being "maybe we don't pass the opctx as an argument". But the test makes sense!

**Comment by zephraph** [comment]
I updated to used the recommended approach (which I just realized was also done in other places).

**Pull request merged**